mllib/src/main/scala/org/apache/spark/mllib/linalg/distributed/RowMatrix.scala

```diff
@@ -384,18 +384,28 @@ class RowMatrix @Since("1.0.0") (
     val n = numCols().toInt
     require(k > 0 && k <= n, s"k = $k out of range (0, n = $n]")
 
-    val Cov = computeCovariance().asBreeze.asInstanceOf[BDM[Double]]
+    if (n > 65535) {
+      val svd = computeSVD(k)
+      val s = svd.s.toArray.map(eigValue => eigValue * eigValue / (n - 1))
```

Comment:
My linear algebra is probably rusty, so check me here, but we need the eigendecomposition of the covariance matrix 1/(n-1) * mat' * mat. I get how the singular values here give these eigenvalues. Don't the singular vectors V need to be divided by n-1 too? I'm probably wrong about it, just checking, esp. as we don't actually have tests for any of the output here!

Reply:
Eigen vectors has unit norm. So scaling doesn't matter. So, we should scale only eigen value matrix.

Actually I have tested the PR against smaller matrix, which is shown in the comment https://github.com/apache/spark/pull/22784#issuecomment-431613820 , by removing the limit check

Reply:
Right, make sense.